### PR TITLE
Create opto-electronic-advances.csl

### PR DESCRIPTION
**Pull request by lancefxh**
This is a CSL citation style for "Opto-Electronic Advances" which can't  be found in the Zotero Style Repository before. The style was created by myself.

### CODE REVIEW

File: opto-electronic-advances.csl [hunk 1]
```diff
@@ -1,10 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
-<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" demote-non-dropping-particle="sort-only" default-locale="en-GB">
+<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" default-locale="en-GB">
   <!-- This style was edited with the Visual CSL Editor (https://editor.citationstyles.org/visualEditor/) -->
   <info>
     <title>Opto-Electronic Advances</title>
     <title-short>Opto-Electron Adv</title-short>
-    <id>http://www.zotero.org/styles/opto-electronic-advances</id>
```
Comment (adam3smith) [inline opto-electronic-advances.csl]: no, you need the id

File: opto-electronic-advances.csl [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        <text variable="title" font-style="italic"/>
+      </if>
+      <else>
+        <text variable="title"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="author">
+    <names variable="author">
+      <name and="symbol" delimiter-precedes-last="never" et-al-min="5" et-al-use-first="5" initialize-with=". " name-as-sort-order="all"/>
```
Comment (adam3smith) [inline opto-electronic-advances.csl:39]: `et-al-min="5" et-al-use-first="5"` is the problem -- `et-al-use-first` needs to be at least one lower then `et-al-min`

Comment (adam3smith) [reply opto-electronic-advances.csl:39]: (and you may want to delete it entirely -- the 6/1 set in the bibliography is more plausible and there's no reason to have it in two places)

Comment (lancefxh) [reply opto-electronic-advances.csl:39]: okay, I will try again

Comment (lancefxh) [reply opto-electronic-advances.csl]: A little difference between paper-conference and others in the suffix ";"and "."

**Comment by rmzelle** [comment]
Thanks! Did you base this on an existing CSL style? If so, we'd like to know which one, so we can add a "template" link (see https://github.com/citation-style-language/styles/wiki/Style-Requirements#8---template-link)

**Comment by rmzelle** [comment]
Thanks!

**Pull request merged**